### ISSUE TITLE
Improve Docker Image extraction speed for linux ec2 instances

### ISSUE THREAD

**Issue by chawinphat** [opensearch-project/opensearch-ci#374]
### Is your feature request related to a problem? Please describe

Docker image extraction can be sped by extracting files in parallel. While Linux extraction time is fast already, parallelizing this process will be a good quality of life change.

### Describe the solution you'd like

We can use [Pigz](https://github.com/madler/pigz) instead of Gzip within the al2023 agent's [scripts].(https://github.com/opensearch-project/opensearch-ci/blob/main/packer/scripts/al2023/al2023-agent-setups.sh)

@peterzhuamazon has already done this implementation for the Windows agent instance in this [code block](https://github.com/opensearch-project/opensearch-ci/blob/main/packer/scripts/windows/scoop-install-commons-docker-support.ps1#L76-L87).

**Comment by jordarlu** [comment]
Thanks for opening this issue and PR, the community willl review it and comment ... thanks again !!

**Comment by peterzhuamazon** [comment]
I will add more changes of pigz soon.
